### PR TITLE
Remove 2nd client hello post openssl 1.1.1 bug workaround

### PR DESCRIPTION
**Pull request by CoolCat467**
In this pull request, we remove a workaround for a bug in OpenSSL 1.1.1 with DTLS `ClientHello`s being eaten. There is a comment noting that it should be safe to remove this workaround in 2024 or so, so the time is neigh!

### CODE REVIEW

File: newsfragments/3097.removal.rst
Comment (A5rocks) [inline newsfragments/3097.removal.rst:1]: I think this is a bit wordy as a newsfragment, maybe it should stick to strictly what changed?

Comment (CoolCat467) [reply newsfragments/3097.removal.rst:1]: Changed in 7f70723

**Review by A5rocks — APPROVED** [review]
I don't know enough about this to say whether it's correct but looks good to me!

**Comment by jakkdl** [comment]
ubuntu 18.06, as mentioned in the comment, reached EOL at 2023-05-31. So yeah I'm guessing it'll be fine

**Comment by CoolCat467** [comment]
Going to merge this soon unless anyone has any additional concerns, consensus seems to be that removing duplicate should be fine.
